From bf018da508057fdb8f946dd1348bae8ab8e33942 Mon Sep 17 00:00:00 2001
From: Nicola Di Lieto <nicola.dilieto@gmail.com>
Date: Sat, 24 Jul 2021 09:43:30 +0200
Subject: [PATCH 157/158] exclude firmware files from 'make clean'

---
 firmware/Makefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/firmware/Makefile b/firmware/Makefile
index 5053f3e4e4..64ef10c9df 100644
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -10,6 +10,13 @@ fw-shipped-$(CONFIG_FW_VRX200_PHY22F_A2X) += lantiq/vrx200_phy22f_a2x.fw
 fw-shipped-$(CONFIG_FW_XRX300_PHY11G_A21) += lantiq/xrx300_phy11g_a21.fw
 fw-shipped-$(CONFIG_FW_XRX300_PHY22F_A21) += lantiq/xrx300_phy22f_a21.fw
 
+no-clean-files += lantiq/vrx200_phy11g_a1x.fw
+no-clean-files += lantiq/vrx200_phy11g_a2x.fw
+no-clean-files += lantiq/vrx200_phy22f_a1x.fw
+no-clean-files += lantiq/vrx200_phy22f_a2x.fw
+no-clean-files += lantiq/xrx300_phy11g_a21.fw
+no-clean-files += lantiq/xrx300_phy22f_a21.fw
+
 fw-shipped-all := $(fw-shipped-y) $(fw-shipped-m) $(fw-shipped-)
 
 quiet_cmd_fwbin = MK_FW   $@
-- 
2.11.0

